apps/backoffice-v2/src/lib/blocks/variants/DefaultBlocks/hooks/useCaseBlocksLogic/hooks/useCaseTabsLogic/useCaseTabsLogic.ts

```diff
@@ -0,0 +1,31 @@
+import { TCaseTabDefinition } from '@/lib/blocks/variants/DefaultBlocks/types/case-tab';
+import { useCallback, useMemo, useState } from 'react';
+
+export const useCaseTabsLogic = (tabs: TCaseTabDefinition[]) => {
+  const [activeTabName, setActiveTabName] = useState<string | null>(() => {
+    return tabs.length ? (tabs.at(0)?.name as string) : null;
+  });
+
+  const activeTab = useMemo(
+    () => tabs?.find(tab => tab.name === activeTabName),
+    [tabs, activeTabName],
+  );
+
+  const setTab = useCallback(
```

Comment:
This way there is no confusion with the state setter and the naming is the same with the addition of "active".
```suggestion
  const handleActiveTab = useCallback(
```